### ISSUE TITLE
Proper service bindings

### ISSUE THREAD

**Issue by hkroger** [monostream/helmi#30]
OSB supports service bindings and creation of those. How could we create service bindings on the fly with Helmi? 

Now it always returns the same. We would like to use it to create different credentials.

**Comment by adrianliechti** [comment]
hey hey,

we designed helmi basically (and obviously :) ) around helm, with the aim to provide a rich service catalog for a CaaS & PaaS, and allow developers and operators to easily create services by them self using "standards", or make benefit from the huge community around public helm charts.
If you have a look at them, they usually allow to specify 1 user using values. this also applies for most of the popular docker images.

if one want to create different credentials/users for every binding, we go into an area where helm is at it's limit - because there is currently no feature to standardise user creation, neither in helm or kubernetes.

the only solution would be to invoke custom code via an API or command from helmi to the service - each one completely different (e.g. cassandra is quite different than e.g. minio or postgres).

a pragmatic solution would be something to "overwrite" the default handling and invoke a script inside a pod to do delegate the job.

I could think of:

chart-hooks:
  bind:
    - command: /scripts/bind.sh
    - args
         {{ ... }}
         {{ ... }}
        

helmi would then send over the json request as first parameter, and additional templated args as the follow ones if specified. it also would need to get back a valid osba json response on stdout - or an errorlevel.

that would allow the script within the pod to create a user (using e.g. a cli-client or curl), format the answer as needed, and handle it over

**Comment by hkroger** [comment]
Yes, that was my guess that it's like this now because of limitations in helm (+charts). That is totally understandable. These extensions or hooks sound like a good way to allow extension of functionality and in this case allow custom service bindings and sounds like it would solve our problem.
